pkg/skaffold/build/misc/env_test.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package misc
+
+import (
+	"testing"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+	"github.com/GoogleContainerTools/skaffold/testutil"
+)
+
+func TestEvaluateEnv(t *testing.T) {
+	tests := []struct {
+		description string
+		env         []string
+		want        []string
+		shouldErr   bool
+	}{
+		{
+			description: "env variables",
+			env:         []string{"key1=value1", "key2=", "key3={{.FOO}}"},
```

Comment:
Should variables be able to refer to other variables in the same level?  For example, `key4={{.key3}}`?

Reply:
Any example why we'd want to do that?

Reply:
I think go templates can be nested? https://golang.org/pkg/text/template/#hdr-Nested_template_definitions We should document any differences from normal usage of go templates, especially for the go-n00bs (like me!) who don't really know the limitations and how these uses would differ from other uses of go templates.

Reply:
I'd rather keep it simple and not go into too much details about go templates.
I added an example to the `buildpacks` sample so that users understand the basic usage.